sys/vm/vm_map.c

```diff
@@ -5155,11 +5231,24 @@ vm_map_growstack(vm_map_t map, vm_offset_t addr, vm_map_entry_t gap_entry)
 		goto retry;
 	}
 
+	/*
+	 * XXX-AM: we need to recreate the capability for the
+	 * stack reservation. This is hacky, we should have
+	 * a way to recover it. Maybe save it as a capability
+	 * in the map entry?
```

Comment:
I think that we may end up having a more complex structure to represent a reservation, instead of a plain vaddr of the start of the reservation. This will allow us to have extra metadata in there, which we also would need to record the original reservation maxprot, depending on how we handle things in `vm_map_delete()`.